### PR TITLE
Introduce dpctl/sycl.pxd

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
This PR adds `dpctl/sycl.pxd` to alleviate the need for Cython extension authors to redeclare SYCL RT classes in their code, and to provide routines to cast from SYCLInterface library's opaque pointers to Sycl class pointers and back. 

Example is added to test and to illustrate the usage of `dpctl.sycl` in Cython.

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to an issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpctl/sycl.pxd [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        pass
+
+    cdef cppclass executable_kernel_bundle \
+        "sycl::kernel_bundle<sycl::bundle_state::executable>":
+        pass
+
+cdef extern from "syclinterface/dpctl_sycl_type_casters.hpp" \
+    namespace "dpctl::syclinterface":
+    # queue
+    cdef dpctl_backend.DPCTLSyclQueueRef wrap_queue "dpctl::syclinterface::wrap<sycl::queue>" (const queue *)
```
Comment (diptorupd) [inline dpctl/sycl.pxd:46]: needs some reformatting

Comment (oleksandr-pavlyk) [reply dpctl/sycl.pxd:46]: You mean like breaking down the line to fit 80 chars?

Comment (diptorupd) [reply dpctl/sycl.pxd:46]: yes

Comment (oleksandr-pavlyk) [reply dpctl/sycl.pxd]: Just pushed.

**Review by diptorupd — APPROVED** [review]

**Pull request merged**